### ISSUE TITLE
Deno run panic & deno lsp server crash when processing this code

### ISSUE THREAD

**Issue by zxch3n** [denoland/deno#11898]
```ts
function imp<T extends string>(x: T): typeof import(T) {
    return x;
}

console.log("123");
```

Error
```
thread 'main' panicked at 'internal error: entered unreachable code: parse_lit should not be called', /Users/lucacasonato/.cargo/registry/src/github.com-1ecc6299db9ec823/swc_ecma_parser-0.66.2/src/parser/expr.rs:1574:18
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Version
```
deno 1.13.2 (release, aarch64-apple-darwin)
v8 9.3.345.11
typescript 4.3.5
```

**Comment by kitsonk** [comment]
That isn't valid TypeScript ([`import(_)` requires a string literal](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABDAtgBwDwBVEFMAeUuYAJgM6JlQBOMYA5gHwAU+AXIlgJQdQCeaXHGDJ0calGbdEAbwBQiRYmq4oIaknwBuOQF85ciAjJwANrgB0puPWYAiAIwAmAMx2uWoA), but it shouldn't panic.

**Comment by kitsonk** [comment]
This has been fixed upstream, and should be included in Deno 1.14.
